src/components/PositionedOverlay/PositionedOverlay.tsx

```diff
@@ -127,14 +128,15 @@ export class PositionedOverlay extends PureComponent<
       fixed,
       preventInteraction,
       classNames: propClassNames,
+      zIndexOverride,
     } = this.props;
 
     const style = {
       top: top == null || isNaN(top) ? undefined : top,
       left: left == null || isNaN(left) ? undefined : left,
       right: right == null || isNaN(right) ? undefined : right,
       width: width == null || isNaN(width) ? undefined : width,
-      zIndex: zIndex == null || isNaN(zIndex) ? undefined : zIndex,
+      zIndex: zIndexOverride || zIndex || undefined,
```

Comment:
Because of `zIndex` typing (`number | null`) we shouldn't need the `isNaN` check? So this simplifies that condition statement I believe.